joomla/joomla-cms
# addHTMLData function

### Description

This PR adds `addHTMLData` public function to JDocumentHTML
This can be used for creanting and adding values to data-\* HTML5 attributes in any HTML element (by tag or by id).
This work in frontend and backend.

Examples of usage:

```
JFactory::getDocument()->addHTMLData('caption', 'img.caption');
JFactory::getDocument()->addHTMLData('caption', 'img.othercaption');
JFactory::getDocument()->addHTMLData('keepalive', array('uri' => '/index.php', 'seconds' => 60000));
JFactory::getDocument()->addHTMLData('somevar', 'somevalue', '#content');
```
### How to test
1. Add one of the examples above somewhere in /libraries/cms/html/behaviour.php. I tested in  `caption` and `keepalive` functions.
2. Go to homepage and check if in the HTML code there is the data-\* attribute you added with the value you added in the element you selected.
### B/C

None

Note: Improvements are welcome
